chapter.md

```diff
@@ -0,0 +1,735 @@
+    this class is used to fill in `<character-form>` elements
+    in the `character-form.html` template snippet.
+*   Using the `@ViewChild` decorator on `ngForm`
+    gives this class an instance variable that watches a form.
+    We need this because we are going to subscribe to event notifications from that form
+    later on.
+*   The `characterForm` instance variable is our working copy of the form's state.
+    FIXME: more explanation needed.
+*   Finally, `private ngRedux: NgRedux<IAppState>` triggers Angular's dependency injection
+    and gives us access to the Redux store.
+    When our application is busy doing other things,
+    our data will live in this store,
+    and when we're testing,
+    we can inject a mock object here to give us more insight.
+
+> The `formSubs` instance variable is the odd one out in this class.
+> Its job is to store the observer/observable subscription
+> connecting our state to our form
+> so that we can unsubscribe cleanly when this component is destroyed.
+> Angular 2 will automatically unsubscribe on our behalf,
+> but it's always safer to put our own toys away when we're done playing with them…
+
+Since `CharacterForm` is an Angular component,
+it needs a template to define how it will be rendered.
+In keeping with good modern practice,
+we will use a template-driven form
+and bind its inputs to objects retrieved from the Redux state.
+The first part of that form looks like this:
+
+```html
+<form #form="ngForm">
+  <label for="name">Character Name:</label>
+  <input
+    type="text"
+    name="name"
+    [(ngModel)]="characterForm.name">
+
+  <label for="age">Age:</label>
+  <input
+    type="number"
+    name="age"
+    [(ngModel)]="characterForm.bioSummary.age">
+</form>
+```
+
+`ngModel` is a magic word meaning "the value of this field in the form".
+The expression `[(ngModel)]` therefore makes a form control
+and binds it to the public data member of our form object.
+The final step in wiring all of this together is therefore to subscribe to the Redux state
+so that whenever it changes,
+the changes are automatically pushed into the form
+(which triggers a DOM update and a re-rendering of the page).
+Equally,
+since we're listening to the form for changes and writing those to our state
+by creating and dispatching actions,
+anything the user does will be reflected in the state.
+The beauty of this is that if anyone else does an update anywhere,
+everything will keep itself in sync.
+
+It's tempting to put this final piece of wiring in the constructor of our component,
+but that doesn't work
+because Angular has to construct several objects
+before any of them can be wired together.
+This is a key feature of Angular's architecture
+(and of the architectures of many other systems):
+object *construction* and object *initialization* are handled separately
+to free us from headaches related to cyclic references.
+
+The right place to connect everything is `ngOnInit`,
```

Comment:
I think that the right place for `ngForm.valueChanges` should be in `ngAfterViewInit`, I will get back to you on this
the `ngRedux.select` for sure is on the right place, but the other one we need to move into a more appropriate lifecycle hook